lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+
+  # Add validation check
+  $app->validator->add_check(foo => sub ($v, $name, $value) {
+    return $value ne 'foo';
+  });
+
+  # Add validation filter
+  $app->validator->add_filter(quotemeta => sub ($v, $name, $value) {
+    return quotemeta $value;
+  });
+
+=head1 METHODS
+
+L<Mojolicious> inherits all methods from L<Mojo::Base> and implements the following new ones.
+
+=head2 build_controller
+
+  my $c = $app->build_controller;
+  my $c = $app->build_controller(Mojo::Transaction::HTTP->new);
+  my $c = $app->build_controller(Mojolicious::Controller->new);
+
+Build default controller object with L</"controller_class">.
+
+  # Render template from application
+  my $foo = $app->build_controller->render_to_string(template => 'foo');
+
+=head2 build_tx
+
+  my $tx = $app->build_tx;
+
+Build L<Mojo::Transaction::HTTP> object and emit L</"after_build_tx"> hook.
+
+=head2 config
+
+  my $hash = $app->config;
+  my $foo  = $app->config('foo');
+  $app     = $app->config({foo => 'bar', baz => 23});
+  $app     = $app->config(foo => 'bar', baz => 23);
+
+Application configuration.
+
+  # Remove value
+  my $foo = delete $app->config->{foo};
+
+  # Assign multiple values at once
+  $app->config(foo => 'test', bar => 23);
+
+=head2 defaults
+
+  my $hash = $app->defaults;
+  my $foo  = $app->defaults('foo');
+  $app     = $app->defaults({foo => 'bar', baz => 23});
+  $app     = $app->defaults(foo => 'bar', baz => 23);
+
+Default values for L<Mojolicious::Controller/"stash">, assigned for every new request.
+
+  # Remove value
+  my $foo = delete $app->defaults->{foo};
+
+  # Assign multiple values at once
+  $app->defaults(foo => 'test', bar => 23);
+
+=head2 dispatch
+
+  $app->dispatch(Mojolicious::Controller->new);
+
+The heart of every L<Mojolicious> application, calls the L</"static"> and L</"routes"> dispatchers for every request
+and passes them a L<Mojolicious::Controller> object.
+
+=head2 handler
+
+  $app->handler(Mojo::Transaction::HTTP->new);
+  $app->handler(Mojolicious::Controller->new);
+
+Sets up the default controller and emits the L</"around_dispatch"> hook for every request.
+
+=head2 helper
+
+  $app->helper(foo => sub {...});
+
+Add or replace a helper that will be available as a method of the controller object and the application object, as well
+as a function in C<ep> templates. For a full list of helpers that are available by default see
+L<Mojolicious::Plugin::DefaultHelpers> and L<Mojolicious::Plugin::TagHelpers>.
+
+  # Helper
+  $app->helper(cache => sub { state $cache = {} });
+
+  # Application
+  $app->cache->{foo} = 'bar';
+  my $result = $app->cache->{foo};
+
+  # Controller
+  $c->cache->{foo} = 'bar';
+  my $result = $c->cache->{foo};
+
+  # Template
+  % cache->{foo} = 'bar';
+  %= cache->{foo}
+
+=head2 hook
+
+  $app->hook(after_dispatch => sub {...});
+
+Extend L<Mojolicious> with hooks, which allow code to be shared with all requests indiscriminately, for a full list of
```

Comment:
It might be worth explicitly stating that multiple hooks can be defined for the same hook type -- i.e. you're not limited to setting a hook just once.